doc/dev/docstring_typehint.md

```diff
@@ -0,0 +1,185 @@
+            https://docs.microsoft.com/azure/cognitive-services/form-recognizer/build-training-data-set
+        :param bool use_training_labels: Whether to train with labels or not. Corresponding labeled files must
+            exist in the blob container if set to `True`.
+        :keyword str prefix: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob URI, use the prefix to restrict sub
+            folders for training.
+        :keyword bool include_subfolders: A flag to indicate if subfolders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed. Not supported if
+            training with labels.
+        :keyword str model_name: An optional, user-defined name to associate with your model.
+        :keyword str continuation_token: A continuation token to restart a poller from a saved state.
+        :return: An instance of an LROPoller. Call `result()` on the poller
+            object to return a :class:`~azure.ai.formrecognizer.CustomFormModel`.
+        :rtype: ~azure.core.polling.LROPoller[~azure.ai.formrecognizer.CustomFormModel]
+        :raises ~azure.core.exceptions.HttpResponseError:
+            Note that if the training fails, the exception is raised, but a model with an
+            "invalid" status is still created. You can delete this model by calling :func:`~delete_model()`
+
+        .. versionadded:: v2.1
+            The *model_name* keyword argument
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sample_train_model_without_labels.py
+                :start-after: [START training]
+                :end-before: [END training]
+                :language: python
+                :dedent: 8
+                :caption: Training a model (without labels) with your custom forms.
+        """
+        ...
+```
+
+The first portion of this docstring is a general description of what the method does. Following the general description of the method is a **required new line** and then documentation for each of the parameters, optional keyword arguments, returned objects, and potentially raised errors.
+
+Required parameters can be documented in one-line or two-lines. Both options can be used in a single docstring for documenting different parameters without issue.
+1. This option works best for parameters that are one of the basic types (`str`, `int`, `bool`, `bytes`, `float`, etc.)
+```python
+:param <type> <param_name>: <Description of the parameter>
+```
+2. This option works best for parameters that use custom models or have many different options
+```python
+:param <param_name>: <Description of the parameter>
+:type <param_name>: <param_type>
+```
+
+Optional keyword arguments can in one-line or two-lines. Both options can be used in a single docstring for documenting different keywords without issue.
```

Comment:
keyword includes kwargs and parameters after the * character if you're on Python 3 code